lib/npg_tracking/glossary/composition/factory.pm

```diff
@@ -0,0 +1,114 @@
+package npg_tracking::glossary::composition::factory;
+
+use Moose::Role;
+use Carp;
+
+use npg_tracking::glossary::composition::component;
+use npg_tracking::glossary::composition;
+
+our $VERSION = '0';
+
+sub create_sequence_component {
+  my $self = shift;
+
+  my $class = 'npg_tracking::glossary::composition::component';
+  my $h = {};
+  for my $attr_obj ( $class->meta->get_all_attributes ) {
+    my $attr = $attr_obj->name;
+    $h->{$attr} = $self->can($attr) && defined $self->$attr ? $self->$attr : undef;
+  }
+  my $component;
```

Comment:
Why are we avoiding the croak (should either position or id_run be missing) here?